src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionConstructorsDeclarationGroupingTest.java

```diff
@@ -0,0 +1,105 @@
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.checks.coding.ConstructorsDeclarationGroupingCheck;
+
+public class XpathRegressionConstructorsDeclarationGroupingTest extends AbstractXpathTestSupport {
+
+    private final Class<ConstructorsDeclarationGroupingCheck> clazz =
+            ConstructorsDeclarationGroupingCheck.class;
+
+    @Override
+    protected String getCheckName() {
+        return clazz.getSimpleName();
+    }
+
+    @Test
+    public void testOne() throws Exception {
+        final File fileToProcess = new File(
+                getPath("InputXpathConstructorsDeclarationGroupingClass.java"));
+
+        final DefaultConfiguration moduleConfig = createModuleConfig(clazz);
+
+        final String[] expectedViolation = {
+            "10:5: " + getCheckMessage(clazz,
+                    ConstructorsDeclarationGroupingCheck.MSG_KEY, 6),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+                "/COMPILATION_UNIT/CLASS_DEF[./IDENT"
+                        + "[@text='InputXpathConstructorsDeclarationGroupingClass']]"
+                        + "/OBJBLOCK/CTOR_DEF[./IDENT"
+                        + "[@text='InputXpathConstructorsDeclarationGroupingClass']]",
+                "/COMPILATION_UNIT/CLASS_DEF[./IDENT"
+                        + "[@text='InputXpathConstructorsDeclarationGroupingClass']]"
+                        + "/OBJBLOCK/CTOR_DEF[./IDENT"
+                        + "[@text='InputXpathConstructorsDeclarationGroupingClass']]"
+                        + "/MODIFIERS",
+                "/COMPILATION_UNIT/CLASS_DEF[./IDENT"
+                        + "[@text='InputXpathConstructorsDeclarationGroupingClass']]"
+                        + "/OBJBLOCK/CTOR_DEF/IDENT"
+                        + "[@text='InputXpathConstructorsDeclarationGroupingClass']"
+
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation, expectedXpathQueries);
+    }
+
+    @Test
+    public void testTwo() throws Exception {
```

Comment:
We usually demand 3 tests for xpath, please add one more for inner class in class or better on https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#COMPACT_CTOR_DEF 

And change name of method avoid abstract Two, and use Enum
. Same for others test methods.

Reply:
Done. Added new test for records. Here it is https://github.com/checkstyle/checkstyle/pull/14749/files#diff-7ed07ebc16b7c739ede9580361207c06573eef922d5c593acac87b48c315ac58

And also renamed the test methods to testClass, testEnum, testRecords.